src/rp2_common/hardware_sync_spin_lock/include/hardware/sync/spin_lock.h

```diff
@@ -19,53 +19,93 @@
 
 // PICO_CONFIG: PICO_SPINLOCK_ID_IRQ, Spinlock ID for IRQ protection, min=0, max=31, default=9, group=hardware_sync
 #ifndef PICO_SPINLOCK_ID_IRQ
+#if PICO_RP2350 && !PICO_USE_SW_SPIN_LOCKS
+#define PICO_SPINLOCK_ID_IRQ 5
+#else
 #define PICO_SPINLOCK_ID_IRQ 9
```

Comment:
@kilograham In the "original" version of this code, do you remember why the "defined" spinlocks start at 9? I.e. is it actually okay for the "modified" version of this code to use spinlocks 5, 6 and 7?

Reply:
i was getting to comment on this; the main reason is that they are used by pico-extras stuff, so yeah, we want to be careful about what we assign into that range (and pick something that isn't likely to be used at runtime) - i just havent spent any brain cycles picking which, but @will-v-pi you are free to ;-)

Reply:
I've opened https://github.com/raspberrypi/pico-extras/pull/99 in pico-extras to throw errors when not using software spin locks, with the error message explaining that you need to define the IDs